metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java

```diff
@@ -177,7 +178,7 @@ synchronized void start(long initialDelay, long period, TimeUnit unit, Runnable
      *
      * Overriding this in a subclass to revert to the old behavior is permitted.
      */
-    protected ScheduledFuture<?> getScheduledFuture(long initialDelay, long period, TimeUnit unit, Runnable runnable) {
+    protected ScheduledFuture<?> getScheduledFuture(long initialDelay, long period, TimeUnit unit, Runnable runnable, ScheduledExecutorService executor) {
         return executor.scheduleWithFixedDelay(runnable, initialDelay, period, unit);
     }
```

Comment:
issue(blocking): This is a breaking change for anybody extending `ScheduledReporter` and overriding `ScheduledReporter#getScheduledFuture(...)`.

Please add a new method instead, delegate this method to the new one, and mark the old method as deprecated.

```java
    /**
     * [...]
     * @deprecated [...]
     */
    @Deprecated
    protected ScheduledFuture<?> getScheduledFuture(long initialDelay, long period, TimeUnit unit, Runnable runnable) {
        return getScheduledFuture(initialDelay, period, unit, runnable, this.executor);
    }

    protected ScheduledFuture<?> getScheduledFuture(long initialDelay, long period, TimeUnit unit, Runnable runnable, ScheduledExecutorService executor) {
        return executor.scheduleWithFixedDelay(runnable, initialDelay, period, unit);
    }